### PR TITLE
zc - Fixed dateConversion test file

### PR DESCRIPTION
**Pull request by imnotmomo**
## Overview
In this PR, I fixed the dateConversion test file that work both locally and remotely. Since there is timezone difference everywhere including my laptop and github action container, I fixed the test to match github action container for the previous PR #13. But after it was being merged to the main, and when we pull from main and run npm test locally, things starting to fail.(the toLocalISOString is fetching the local timezone which will be used to calculate offset)
The new test file included manually imported timezone mock and tz database, which mock the timezone with ('US/Paccific'), so the test file can be executed correctly in anywhere on earth now...

## Tests
- Backend Unit tests (`mvn test`) pass
- Backend Test coverage (`mvn test jacoco:report`) 100%
- Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- Frontend Unit tests (`npm test`) pass
- Frontend Test coverage (`npm run coverage`) 100%
- Frontend Mutation tests (`npx stryker run`) 100% 
- Frontend Linting (`npx eslint --fix src`) 

## Linked Issues
None, but related to PR #13

### CODE REVIEW

**Review by jamesyfu — DISMISSED** [review]
Great job Chloe, LGTM.

File: frontend/src/tests/components/Utils/DateConversion.test.js [hunk 1]
```diff
@@ -1,42 +1,378 @@
...
+
+function MockDate(param) {
+  if (arguments.length === 0) {
+    this.d = new _Date();
+  } else if (arguments.length === 1) {
+    if (param instanceof MockDate) {
+      this.d = new _Date(param.d);
+    } else if (typeof param === 'string') {
+      if (param.match(date_iso_8601_regex) ||
+        param.match(date_with_offset) ||
```
Comment (pconrad) [inline frontend/src/tests/components/Utils/DateConversion.test.js]: This seems like a lot of custom code to do something that can probably be done by a library that's already tested, and is maintained by someone else.

What is the real problem you are trying to solve?

**Review by pconrad — CHANGES_REQUESTED** [review]
This seems like a much, much more complicated approach than is necessary.

Typically, we just find a library routine that does what we need.

If necessary, you can add that to package.json if it's not already there.

What problem are you really trying to solve with all of this code?

File: frontend/src/tests/components/Utils/DateConversion.test.js (hunk 1 above)
Comment (pconrad) [reply frontend/src/tests/components/Utils/DateConversion.test.js]: See also: https://ucsb-cs156-s24.slack.com/archives/C06RPBBPPLJ/p1716606773975469

Comment (pconrad) [reply frontend/src/tests/components/Utils/DateConversion.test.js]: And: https://bugfender.com/blog/javascript-date-and-time/

Comment (imnotmomo) [reply frontend/src/tests/components/Utils/DateConversion.test.js]: My code is fetching the environment’s timezone for testing. And the expected date/time is a fixed number when made the test. The thing is my local machine, GitHub action container are on two different timezone, which means the result it get will not match the number I put in the expected field. In my previous test, I try to match this number from the result I got from GitHub action test. But after that branch is being merged to main, and when we pulled from main to make our new branch, the local tests start to failing for date conversion. What I’m trying to do is fixing a timezone for the test file, and I did not know I could import a package by adding it into the package.json file. So I just manually added the tz database and required function for the mocking.

Comment (imnotmomo) [reply frontend/src/tests/components/Utils/DateConversion.test.js]: I’ll modify relevant files to import the timezone-mock package to complete this test.

**Review by YixinXue — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM!  10 points for taking the initiative to find this.

**Pull request merged**